OIIInspector/ImageManager.py

```diff
@@ -0,0 +1,151 @@
+import logging
+from retry import retry
+import subprocess
+import time
+import shlex
+from contextlib import suppress
+from OIIInspector.Exceptions import AddressAlreadyInUse, OIIInspectorError
+from OIIInspector.utils import run_cmd
+
+log = logging.getLogger(__name__)
+
+
+class ImageManager:
+    _grpc_start_port = 50051
+    _grpc_max_port_tries = 100
+    _grpc_max_tries = 5
+    _grpc_init_wait_time = 1000
+    _base_container_name = "OIIInspector_running_image"
+
+    def __init__(self):
+        self._container_platform = self._get_container_platform()
+        self._image_address = None
+        self._port = self._grpc_start_port
+        self._rpc_proc = None
+
+    def start_image(self, image_address):
+        self._image_address = image_address
+        self._remove_old_image_if_exists()
```

Comment:
Is this a good idea? All instances of this class have the same start port upon construction. So if two objects run in parallel and run two different containers, the containers will have the same name and second object will remove the container of the first object.